xray/src/build_quadtree.rs

```diff
@@ -174,8 +174,11 @@ pub fn run<T: Extension>(data_provider_factory: DataProviderFactory) {
         .unwrap()
         .map(String::from)
         .collect::<Vec<_>>();
-    let point_cloud_client = PointCloudClient::new(&point_cloud_locations, data_provider_factory)
-        .expect("Could not create point cloud client.");
+    let mut point_cloud_client =
```

Comment:
nit: how about passing `number of threads` as a new, maybe optional, parameter to `PointCloudClient::new`?

Reply:
I thought about this, but I would prefer changing the interface in a separate PR, since there are already 3 users.

Reply:
ah, okay